### DISCUSSION TITLE
`http_server` accepting OPTIONS method

### DISCUSSION THREAD

**Discussion by ijsnow** [vectordotdev/vector#16219] [General]
Hello! 

Is there a reason the `http_server` source doesn't handle the `OPTIONS` method when a request is coming in?

I'm tinkering with vector and am trying to send info in the browser to vector through a library and that request starts with `OPTIONS` which fails and vector outputs this info:

```
2023-01-31T19:07:08.625049Z ERROR vector::internal_events::http: Internal error. error_type="connection_failed" stage="receiving" internal_log_rate_limit=true
2023-01-31T19:07:08.650134Z ERROR vector::internal_events::http: Internal log [Internal error.] is being rate limited.
```

I'm able to send POST through Postman with my config but not from the browser because of this.

**Comment by davidhuie-dd** [comment]
That source only handles a single HTTP method at a time: https://vector.dev/docs/reference/configuration/sources/http_server/#method

If supporting OPTIONS is important to you, please feel free to make an issue and we'll take a look. I suspect that in typical Vector setups, OPTIONS requests aren't used since you don't need to discover the accepted methods; you have the config file which explicitly says what's accepted 😄

**Reply by ijsnow** [reply]
Guess you're saying the typical setup is ingesting info from a more controlled environment than the browser with forwarding or something? :P

I asked because it doesn't seem like it'd be difficult to respond with the allowed method, but other CORS concepts like allowed origins might be more complex.
